pkg/controller/operator/seed/resources/kubermatic/admission.go

```diff
@@ -90,6 +90,58 @@ func ClusterAdmissionWebhookCreator(cfg *operatorv1alpha1.KubermaticConfiguratio
 	}
 }
 
+func ClusterMutatingWebhookConfigurationCreator(cfg *operatorv1alpha1.KubermaticConfiguration, client ctrlruntimeclient.Client) reconciling.NamedMutatingWebhookConfigurationCreatorGetter {
+	return func() (string, reconciling.MutatingWebhookConfigurationCreator) {
+		return ClusterAdmissionWebhookName, func(hook *admissionregistrationv1.MutatingWebhookConfiguration) (*admissionregistrationv1.MutatingWebhookConfiguration, error) {
+			matchPolicy := admissionregistrationv1.Exact
+			failurePolicy := admissionregistrationv1.Fail
+			sideEffects := admissionregistrationv1.SideEffectClassNone
```

Comment:
I'm not sure is this the correct policy for our webhook. I've checked the [official docs](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#side-effects), but I'm not really sure I understand this at all.

I'm not sure what they mean by "out-of-band changes". To me, this reads as changing some other/different objects than the object that we're mutating, but I'm not sure. I can look a bit more into this, but suggestions and feedback are appreciated.

Reply:
Yes, I also understand it that way. Our mutating webhook is "side-effects" free as we are not changing any state in the scope of the cluster admission. LGTM